Preface.Rmd

```diff
@@ -1,116 +1,115 @@
     loops!
 
 *   Chapter \@ref(functions), "Functions", has an improved ordering,
-    introduces the pipe (`%>%`) as a third way to compose functions (Section 
-    \@ref(function-composition)), and has considerably improved coverage of 
+    introduces the pipe (`%>%`) as a third way to compose functions (Section
+    \@ref(function-composition)), and has considerably improved coverage of
     function forms (Section \@ref(function-forms)).
 
 *   Chapter \@ref(environments), "Environments", has a reorganised treatment of
-    special environments (Section \@ref(special-environments)), and a much 
+    special environments (Section \@ref(special-environments)), and a much
     improved discussion of the call stack (Section \@ref(call-stack)).
 
-*   Chapter \@ref(conditions), "Conditions", contains material previously 
+*   Chapter \@ref(conditions), "Conditions", contains material previously
     in "Exceptions and debugging", and much new content on how R's condition
     system works. It also shows you how to create your own custom condition
     classes (Section \@ref(custom-conditions)).
 
-The chapters following foundations have been re-organised around the three most important programming paradigms in R: functional programming, object oriented programming, and metaprogramming.
+The chapters following the Part I Foundations segment have been re-organised around the three most important programming paradigms in R: functional programming, object oriented programming, and metaprogramming.
 
-* Functional programming is now more cleanly divided into the three main 
-  techniques: "Functionals" (Chapter \@ref(functionals)), "Function 
+* Functional programming is now more cleanly divided into the three main
+  techniques: "Functionals" (Chapter \@ref(functionals)), "Function
   factories" (Chapter \@ref(function-factories)), and "Function operators"
   (Chapter \@ref(function-operators)). I've focussed in on ideas that have
   practical applications in data science and reduced the amount of pure theory.
-  
-  These chapters now use functions provided by the purrr package [@purrr], 
-  which allow me to focus more on the underlying ideas and less on the 
-  incidental details. This lead to a considerable simplification of the 
-  function operators chapter since a major use was to work around the absence 
-  of `...` in base functionals.
-  
-* Object oriented programming (OOP) now forms a major section of the book with 
-  completely new chapters on base types (Chapter \@ref(base-types)), 
-  S3 (Chapter \@ref(s3)), S4 (Chapter \@ref(s4)), R6 (Chapter \@ref(r6)), 
+
+  These chapters now use functions provided by the purrr package [@purrr],
+  which allow me to focus more on the underlying ideas and less on the
+  incidental details. This led to a considerable simplification of the
+  function operators chapter since a major use was to work around the absence
+  of ellipses [`...`] in base functionals.
```

Comment:
Should these be `()`?

Reply:
Yes, of course. The editor used [ ] but I think that ( ) is more reasonable. I'll keep this in mind.